File: Comp_management.py
class Employee:
    def __init__(self, name, work_days, salary_per_day):
        self.name = name
        self.work_days = work_days
        self.salary_per_day = salary_per_day

    def compute_salary(self):
        return self.work_days * self.salary_per_day

    def info(self):
        return f"Name: {self.name}, WorkDays: {self.work_days}, Salary: {self.compute_salary()}"


class Department:
    def __init__(self, dept_name):
        self.dept_name = dept_name
        self.employees = []


class Company:
    def __init__(self, name):
        self.name = name
        self.departments = []
        self.manager = None
        self.deputy_managers = []
        self.employees = []

    def total_salary(self):
        return sum(e.compute_salary() for e in self.employees)

    def max_work_days_employee(self):
        if not self.employees:
            return None
        return max(self.employees, key=lambda e: e.work_days)

    def sort_by_salary_descending(self):
        return sorted(self.employees, key=lambda e: e.compute_salary(), reverse=True)

    def employees_with_work_days_less_than_30(self):
        return [e for e in self.employees if e.work_days < 30]


def main():
    company_name = input("Enter the company name: ")
    comp = Company(company_name)

    amount_e = int(input("The amount of employees: "))

    for i in range(amount_e):
        print(f"\nEmployee {i + 1}:")
        name = input("Name: ")
        work_days = int(input("Work days: "))
        salary_per_day = float(input("Salary per day: "))

        emp = Employee(name, work_days, salary_per_day)
        comp.employees.append(emp)

    print(f"\nTotal salary of company: {comp.total_salary()}")

    max_emp = comp.max_work_days_employee()
    if max_emp:
        print(f"Employee with max work days: {max_emp.info()}")

    print("\nEmployees sorted by salary (descending):")
    for e in comp.sort_by_salary_descending():
        print(e.info())

    print("\nEmployees with work days < 30:")
    for e in comp.employees_with_work_days_less_than_30():
        print(e.info())


if __name__ == "__main__":
    main()

File: Comp_management.java
import java.util.*;

class Employee {
    private String name;
    private int workDays;
    private double salaryPerDay;

    public Employee(String name, int workDays, double salaryPerDay) {
        this.name = name;
        this.workDays = workDays;
        this.salaryPerDay = salaryPerDay;
    }

    public String getName() {
        return name;
    }

    public int getWorkDays() {
        return workDays;
    }

    public double getSalaryPerDay() {
        return salaryPerDay;
    }

    public double computeSalary() {
        return workDays * salaryPerDay;
    }

    public String info() {
        return "Name: " + name + ", WorkDays: " + workDays + ", Salary: " + computeSalary();
    }
}

class Department {
    private String deptName;
    private List<Employee> employees;

    public Department(String deptName) {
        this.deptName = deptName;
        this.employees = new ArrayList<>();
    }

    public String getDeptName() {
        return deptName;
    }

    public List<Employee> getEmployees() {
        return employees;
    }
}

class Company {
    private String name;
    private List<Department> departments;
    private Employee manager;
    private List<Employee> deputyManagers;
    private List<Employee> employees;

    public Company(String name) {
        this.name = name;
        this.departments = new ArrayList<>();
        this.deputyManagers = new ArrayList<>();
        this.employees = new ArrayList<>();
    }

    public String getName() {
        return name;
    }

    public List<Employee> getEmployees() {
        return employees;
    }

    public double totalSalary() {
        double sum = 0;
        for (Employee e : employees) {
            sum += e.computeSalary();
        }
        return sum;
    }

    public Employee maxWorkDaysEmployee() {
        if (employees.isEmpty()) return null;
        return Collections.max(employees, Comparator.comparingInt(Employee::getWorkDays));
    }

    public List<Employee> sortBySalaryDescending() {
        List<Employee> sorted = new ArrayList<>(employees);
        sorted.sort((e1, e2) -> Double.compare(e2.computeSalary(), e1.computeSalary()));
        return sorted;
    }

    public List<Employee> employeesWithWorkDaysLessThan30() {
        List<Employee> result = new ArrayList<>();
        for (Employee e : employees) {
            if (e.getWorkDays() < 30) {
                result.add(e);
            }
        }
        return result;
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the company name: ");
        String companyName = sc.nextLine();
        Company comp = new Company(companyName);

        System.out.print("The amount of employees: ");
        int amount_e = Integer.parseInt(sc.nextLine());

        for (int i = 0; i < amount_e; i++) {
            System.out.println("\nEmployee " + (i + 1) + ":");
            System.out.print("Name: ");
            String name = sc.nextLine();
            System.out.print("Work days: ");
            int workDays = Integer.parseInt(sc.nextLine());
            System.out.print("Salary per day: ");
            double salaryPerDay = Double.parseDouble(sc.nextLine());

            Employee emp = new Employee(name, workDays, salaryPerDay);
            comp.getEmployees().add(emp);
        }

        System.out.println("\nTotal salary of company: " + comp.totalSalary());

        Employee maxEmp = comp.maxWorkDaysEmployee();
        if (maxEmp != null) {
            System.out.println("Employee with max work days: " + maxEmp.info());
        }

        System.out.println("\nEmployees sorted by salary (descending):");
        for (Employee e : comp.sortBySalaryDescending()) {
            System.out.println(e.info());
        }

        System.out.println("\nEmployees with work days < 30:");
        for (Employee e : comp.employeesWithWorkDaysLessThan30()) {
            System.out.println(e.info());
        }

        sc.close();
    }
}
